source-mongodb/pull.go

```diff
@@ -220,6 +220,12 @@ func (c *capture) ChangeStream(ctx context.Context, client *mongo.Client, resour
 
 		var startAtWithSafety = streamStartAt.Add(oplogSafetyBuffer)
 		if err := oplogHasTimestamp(ctx, client, startAtWithSafety); err != nil {
+			cp, err := json.Marshal(captureState{})
```

Comment:
I believe this is where we would end up if the backfill took so long that the required position in the oplog is no longer available. In this case I think we'd want to not reset the checkpoint, and just error out, since re-doing the backfill will likely end up just repeating and failing again.

Reply:
@williamhbaker I think the issue is we can't be sure that's exactly what is happening here, other cases could be happening if we are unable to find the timestamp in the oplog, for example: the backfill was done, we started a change stream on a single database (not the instance), but there were no events for a while (an idle database). This means we don't have a resume token yet.

The connector is then restarted and we try to start the change stream from the timestamp again, but this time we can't find the timestamp in the oplog anymore.

Reply:
Hmm I see. This seems fine for now since we haven't seen anything close to not being able to backfill in ~24 hrs, but might be good to keep an eye on going forward.